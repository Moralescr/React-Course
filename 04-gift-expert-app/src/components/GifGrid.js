import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs = async () => {

        const apiKey = '7OCSJVhkN01ZLCaCAwaMxN9U4fwXVbiP';
        const url = `api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=${apiKey}`;

        const resp = await fetch(url); //Realiza peticiÃ³n
        const { data } = await resp.json(); //Recupera la data
        
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.downsized_medium.url
            }
        });
    }

    return (
        <div>
          <h3>{category}</h3>
        </div>
    )
}
