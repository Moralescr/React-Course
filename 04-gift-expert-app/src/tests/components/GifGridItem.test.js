import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en GifgridItem', () => {

    const title = 'Un título';
    const url = 'https://localhost/algo';
    const wrapper = shallow(<GifGridItem title = {title} url = {url}/>);


    test('Pruebas de mostrar componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })


    test('Debe de tener un párrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);  
    })


    test('Debe de tener la imagen igual al del url y alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url );  
        expect(img.prop('alt')).toBe( title );  
    })

    test('Debe de tener el animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
});