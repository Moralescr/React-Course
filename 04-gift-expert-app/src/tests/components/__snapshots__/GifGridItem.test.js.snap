// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifgridItem Pruebas de mostrar componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo"
  />
  <p>
    Un título
  </p>
</div>
`;
